tests/Http/HttpJsonResponseTest.php

```diff
@@ -4,72 +4,64 @@
 
 use JsonSerializable;
 use PHPUnit\Framework\TestCase;
+use Illuminate\Http\JsonResponse;
 use Illuminate\Contracts\Support\Jsonable;
 use Illuminate\Contracts\Support\Arrayable;
 
 class HttpJsonResponseTest extends TestCase
 {
-    public function testSetAndRetrieveJsonableData()
-    {
-        $response = new \Illuminate\Http\JsonResponse(new JsonResponseTestJsonableObject);
-        $data = $response->getData();
-        $this->assertInstanceOf('stdClass', $data);
-        $this->assertEquals('bar', $data->foo);
-    }
-
-    public function testSetAndRetrieveJsonSerializeData()
+    /**
+     * @dataProvider setAndRetrieveDataProvider
+     *
+     * @param  $data
+     */
+    public function testSetAndRetrieveData($data): void
     {
-        $response = new \Illuminate\Http\JsonResponse(new JsonResponseTestJsonSerializeObject);
-        $data = $response->getData();
```

Comment:
I think this variable should stay there $response->getData() could be expensive, you are calling it 2 times in the same block.